utils/dyn-config.c

```diff
@@ -569,22 +594,21 @@ glusterBlockDynConfigStart(void *arg)
         continue;
       }
 
-      /*
-       * If force to write to the unwritable or crashed
-       * config file, the vi/vim will try to move and
-       * delete the config file and then recreate it again
-       * via the *.swp
-       */
-      if ((event->mask & IN_IGNORED) && !access(cfg->configPath, F_OK)) {
-        wd = inotify_add_watch(monitor, cfg->configPath, IN_ALL_EVENTS);
+      /* If stat fails we will skip the modify time check */
+      if (!stat(cfg->configPath, &statbuf)) {
+          if (statbuf.st_mtim.tv_sec == mtim.tv_sec &&
+              statbuf.st_mtim.tv_nsec == mtim.tv_nsec) {
+              continue;
+          }
       }
 
+      mtim.tv_sec = statbuf.st_mtim.tv_sec;
+      mtim.tv_nsec = statbuf.st_mtim.tv_nsec;
```

Comment:
should lines 597-606 be under critical section ?

Reply:
This will be okay and no need here.

Reply:
OK